app/uk/gov/hmrc/gform/auditing/AuditService.scala

```diff
@@ -102,10 +103,10 @@ trait AuditService {
         case mr: AuthenticatedRetrievals =>
           Json.toJson(
             Map(
-              "nino"     -> mr.getTaxIdValue(None, "NINO"),
```

Comment:
Can you use the field values in the corresponding ServiceNameAndTaxId values here instead of the Option values & literal strings?

Reply:
I have removed "literal" introducing an ADT - are commenting on the removed line?

Reply:
It's more that there's this in PrepropService:
```scala
    case SaUtr                  => retrievals.getTaxIdValue(Some("IR-SA"), "UTR")
    case CtUtr                  => retrievals.getTaxIdValue(Some("IR-CT"), "UTR")
```

& this in AuditService:
```scala
              "saUtr"    -> mr.getTaxIdValue(Some("IR-SA"), "UTR"),
              "ctUtr"    -> mr.getTaxIdValue(Some("IR-CT"), "UTR"),
```

Where the right hand sides are identical. Seems like some sort of abstraction should be possible, even if it's just putting the (Some("IR-SA"), "UTR"), etc, somewhere common.

It's also a bit bothersome that the full set of cases in these two blocks of code intersect, but are not identical. Is that expected?